docs/proposals/003-model-server-protocol/protocol.md

```diff
@@ -15,63 +15,47 @@ request body, such as the `model` field.
 
 The inference extension scrapes metrics from the model servers to make optimal request scheduling
-decisions. The model servers SHOULD provide the following metrics via a Prometheus endpoint. While
+decisions. The model servers MUST provide the following metrics via a Prometheus endpoint. While
 the metric names may differ slightly in different model servers, the metric types MUST be the same.
 We will align with the
 [model server metrics standardization](https://docs.google.com/document/d/1SpSp1E6moa4HSrJnS4x3NpLuj88sMXr2tbofKlzTZpk)
 effort.
 
 We also show the metrics in vLLM, which is already integrated into the inference extension. We will
-add more model server once they are integrated.
+add more model servers once they are integrated.
 
 | Metric | Type | Description | vLLM metric |
 | ----- | ---- | ---- | ---- |
 | TotalQueuedRequests         | Gauge     | The current total number of requests in the queue.| `vllm:num_requests_waiting`|
 | KVCacheUtilization| Gauge     | The current KV cache utilization in percentage.| `vllm:gpu_cache_usage_perc`|
 
-
-### Future Metrics
-The following metrics MAY be needed in the future for further optimization.
-
-| Metric |Type | Description | vLLM   metric |
-| ----- | ---- | ---- | ---- |
-| TotalTokensInCurrentBatch   | Gauge     | Number of tokens in the current batch.| `vllm:num_tokens_running`|
-| TotalQueuedTokens| Gauge     | The current total number of tokens in the queued requests.| `vllm:num_tokens_waiting` (need to be added)|
-| MaxTokenCapacity| Gauge     | The total size of the KV cache in number of tokens.| `vllm:max_token_capacity` <br> NOTE: This info is available indirectly in [`cache_config_info`](https://github.com/vllm-project/vllm/blob/15702038642192002cd8973cf8948751b750fd07/vllm/engine/metrics.py#L551) metric already , and also  proposed in [model server metrics standardization](https://docs.google.com/document/d/1SpSp1E6moa4HSrJnS4x3NpLuj88sMXr2tbofKlzTZpk). |
-| TimePerPrefillToken | Histogram | The prefill latency per token in the last W seconds. W will be decided by simulation/benchmarking.  In time series metric the latency is typically reported as Histogram and we can derive the average from the Histogram. | `vllm:time_to_first_token_seconds` | 
-| TimePerDecodeToken | Histogram | The decode latency per token in the last W seconds. W will be decided by simulation/benchmarking. | `vllm:time_per_output_token_seconds` | 
-
-## LoRA Adapter Serving
-
-### Dynamic LoRA Serving
+## [Experimental] LoRA Adapter Serving
 
 Model servers that support dynamic LoRA serving can gain additional benefit from the inference
-extension's LoRA affinity algorithm. While dynamic LoRA serving is quite new and evolving, and there
-is no common standard, the inference extension generally expects the following behavior.
+extension's LoRA affinity algorithm. As dynamic LoRA serving is quite new and evolving, this part is considered experimental and subject to changes in future releases.
```

Comment:
See my comment above about following the Gateway API [conformance](https://gateway-api.sigs.k8s.io/concepts/conformance/?h=conform) definitions.